### ISSUE TITLE
Silent fail on connect() if IP changed or no network

### ISSUE THREAD

**Issue by seriema** [studioimaginaire/phue#49]
Things were working fine untill today when suddenly my app would crash when setting the lights.

Running `phue.py` gave this:

````
$ python phue.py --host 192.168.1.3
INFO:phue:Attempting to connect to the bridge...
INFO:phue:Using ip: 192.168.1.3
INFO:phue:Using username from config: [censored]
````

Seems ok, but those messages are misleading...  If I check the source code it's obvious why:

````
    def connect(self):
        """ Connect to the Hue bridge """
        logger.info('Attempting to connect to the bridge...')
        # If the ip and username were provided at class init
        if self.ip is not None and self.username is not None:
            logger.info('Using ip: ' + self.ip)
            logger.info('Using username: ' + self.username)
            return
````

So in my app `b = Bridge('192.168.1.3')` would run fine, but `b.get_light_objects('id')` would fail with exception: `[Errno 61] Connection refused`.

I logged into meethue.com and noticed my bridge was now on `192.168.1.2`. Changing to that I got `PhueRegistrationException` which is good. But I'd expect to get that or some other exception when running `Bridge(ip)`. Otherwise it's hard for me to do proper error handling for my users.

I haven't tried the undocumented methods of registering, like `Bridge::register_app()` or `Bridge::get_ip_address()` but it doesn't seem like they would help as once the IP and username are set `phue` will always assume a connection.

**Comment by natcl** [comment]
Well the idea is to reserve a fixed IP for your bridge on your router, this makes things much easier....
Otherwise you can just store the key yourself and call it when you call the Bridge object:
b = Bridge(ip, key)

Also over here if I call b = Bridge('some_random_ip') I get phue.PhueRequestTimeout not sure why that's not working on your side...

**Comment by natcl** [comment]
Hmm I think I get understand better now.
It will fail silently if you attempt to connect to an ip that has already been registered before.
Perhaps there could be a check in bridge.connect() to see if it responds.

**Comment by seriema** [comment]
Yes exactly, I expect `connect()` to let me know if it actually connect or not. A bonus would be if it can tell me the difference between:
* a general connection error (no wifi etc)
* the bridge it expected isn't reachable (IP changed or wrong network)
* bridge asking for re-validation despite previously validated (maybe because of firmware update or something)

So basically, just because `connect()` has worked once doesn't mean that future connections can be assumed to work just because the local phue config was found.

I'm using phue to make a little app so I'd like to inform my users of any issues and without requiring fixed IP or technical/programming skills from them.

**Comment by redaxmedia** [comment]
I came across this issue while using threading that caused multiple calls of `Bridge(ip)` and therefore multiple instances also.

Store the `Bridge` instance to an global variable and add an condition if the variable is None.

Of course I would call this a back or let's say a wrong type of exception and message.
